src/core_plugins/kibana/public/management/sections/settings/advanced_import_state.js

```diff
@@ -0,0 +1,39 @@
+// State store for managing the state involved with importing advanced settings
+const AdvancedImportState = () => {
```

Comment:
I think one-lining exports is easier to read, like so: `export const AdvancedImportState = () => {`

The revealing pattern later on makes this a bit harder to read though. You're not using scope/`this` anywhere, you could just export an object.

```js
return {
  isImporting() { return importing },
  setImporting(value) { importing = value; },
  addSetting(name, value, oldValue, config, isValid) {
    ...
}
```